controller/eks-cluster-config-handler.go

```diff
@@ -495,9 +497,17 @@ func (h *Handler) create(config *eksv1.EKSClusterConfig, sess *session.Session,
 		}
 	}
 
-	config = config.DeepCopy()
-	config.Status.Phase = eksConfigCreatingPhase
-	return h.eksCC.UpdateStatus(config)
+	err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		config, err = h.eksCC.Get(config.Namespace, config.Name, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		config.Status.Phase = eksConfigCreatingPhase
+		config.Status.FailureMessage = ""
+		config, err = h.eksCC.UpdateStatus(config)
+		return err
+	})
+	return config, err
```

Comment:
After reading the description the use of `RetryOnConflict` makes sense. I think tagging the cluster so we know it belongs to rancher may be an alternative but that complicates things since the user can change tags, we would have to make sure that one is preserved and probably just not display it. If we stick with this can you add a comment on top explaining why it is used here? I think this function should be used sparingly for exceptions like this one and the justification should be clear.